src/ostree/ot-builtin-commit.c

```diff
@@ -95,6 +97,7 @@ static GOptionEntry options[] = {
   { "no-xattrs", 0, 0, G_OPTION_ARG_NONE, &opt_no_xattrs, "Do not import extended attributes", NULL },
   { "link-checkout-speedup", 0, 0, G_OPTION_ARG_NONE, &opt_link_checkout_speedup, "Optimize for commits of trees composed of hardlinks into the repository", NULL },
   { "tar-autocreate-parents", 0, 0, G_OPTION_ARG_NONE, &opt_tar_autocreate_parents, "When loading tar archives, automatically create parent directories as needed", NULL },
+  { "tar-pathname-filter", 0, 0, G_OPTION_ARG_STRING, &opt_tar_pathname_filter, "When loading tar archives, use REGEX,REPLACEMENT against path names", "REGEX,REPLACEMENT" },
```

Comment:
This feels like it should be a `G_OPTION_ARG_STRING_ARRAY` instead, where the first filter that applies wins. Though since it's fully backwards compatible, we can do that later if we feel there's a use case for having CLI support for it.

Reply:
Agreed; let's consider this later, in the end I think we want to encourage use of the API.